modules/aws-databricks-unity-catalog/main.tf

```diff
@@ -8,23 +13,11 @@ resource "databricks_metastore" "this" {
 
 resource "databricks_metastore_data_access" "this" {
   metastore_id = databricks_metastore.this.id
-  name         = aws_iam_role.metastore_data_access.name
+  name         = local.iam_role_name
   aws_iam_role {
-    role_arn = aws_iam_role.metastore_data_access.arn
+    role_arn = local.iam_role_arn
   }
   is_default = true
-  depends_on = [
```

Comment:
I think that it was added because of the role propagation delay - we don't have it anymore?

Reply:
In this version, the metastore_data_access resource is created before the role. It's needed so that the external ID is available so that we can include it in the assume role policy. I probably need to add `skip_validation` here so that we don't verify the created IAM role (since it doesn't exist yet).